src/main/java/jep/SubInterpreterOptions.java

```diff
@@ -0,0 +1,195 @@
+/**
+ * Copyright (c) 2023 JEP AUTHORS.
+ *
+ * This file is licensed under the the zlib/libpng License.
+ *
+ * This software is provided 'as-is', without any express or implied
+ * warranty. In no event will the authors be held liable for any
+ * damages arising from the use of this software.
+ * 
+ * Permission is granted to anyone to use this software for any
+ * purpose, including commercial applications, and to alter it and
+ * redistribute it freely, subject to the following restrictions:
+ * 
+ *     1. The origin of this software must not be misrepresented; you
+ *     must not claim that you wrote the original software. If you use
+ *     this software in a product, an acknowledgment in the product
+ *     documentation would be appreciated but is not required.
+ * 
+ *     2. Altered source versions must be plainly marked as such, and
+ *     must not be misrepresented as being the original software.
+ * 
+ *     3. This notice may not be removed or altered from any source
+ *     distribution.
+ */
+package jep;
+
+/**
+ * A configuration object containing additional options for constructing a SubInterpreter.
+ *
+ * These options map directly to the options documented at
+ * https://docs.python.org/3/c-api/init.html#c.PyInterpreterConfig.
+ * 
+ * These options are ignored if a Python version earlier than 3.12 is used with Jep.
+ *
+ * @since 4.2
+ */
+public class SubInterpreterOptions {
+
+    protected boolean isolated;
+
+    protected int useMainObmalloc = -1;
+
+    protected int allowFork = -1;
+
+    protected int allowExec = -1;
+
+    protected int allowThreads = -1;
+
+    protected int allowDaemonThreads = -1;
+
+    protected int checkMultiInterpeExtensions = -1;
+
+    protected int ownGIL = -1;
+
+    protected SubInterpreterOptions(boolean isolated){
+        this.isolated = isolated;
+    }
+
+    /**
+     * If this is false then the sub-interpreter will use its own "object"
+     * allocator state. Otherwise it will use (share) the main interpreter’s.
+     * 
+     * If this is false then check_multi_interp_extensions must be true. If this
+     * is true then gil must not be true.
+     *
+     * Shared Modules cannot be used when this is set to true.
+     *
+     * @param useMainObmalloc
+     *            whether the sub-interpreter will share the main interpreter's
+     *            allocator state.
+     * @return a reference to this SubInterpreterOptions
+     *
+     */
+    public SubInterpreterOptions setUseMainObmalloc(boolean useMainObmalloc) {
+        this.useMainObmalloc = useMainObmalloc ? 1 : 0;
+        return this;
+    }
+
+    /**
+     * If this is 0 then the runtime will not support forking the process in
```

Comment:
I realize you probably pulled the wording straight from the Python docs, but I think since it's a boolean it should be "If this is true".

Reply:
Fixed in 4f2db10